### DISCUSSION TITLE
Unable to invoke commands such as 'ha core info' on remote (HA)

### DISCUSSION THREAD

**Discussion by mikael-andersson** [hassio-addons/addon-ssh#373] [Q&A]
# Problem/Motivation

Unable to invoke commands such as 'ha core info' with SSH.

## Expected behavior

I expect to be able to run the 'ha' commands successfully as I would when I login via SSH.

## Actual behavior

When running e.g. 'ssh root@MY-HA-HOSTNAME ha core info' and other 'ha' commands, I get:

ERRO[0000] Unexpected server response. Status code: 401 
Unexpected server response. Status code: 401

## Steps to reproduce

Run e.g. the following from your PC:

'ssh root@MY-HA-HOSTNAME ha core info'

## Proposed changes

NA

**Comment by sinclairpaul** [comment]
This is a non interactive/interactive thing I believe, source the profile.

Edit - It's not an addon issue, just SSH, converting to discussion.

**Reply by mikael-andersson** [reply]
@sinclairpaul I forgot one thing in my issue, namely that I do not get an error message when running e.g. 'ha help'. Why do I get an error message when running 'ha core info' and not 'ha help'?

**Reply by sinclairpaul** [reply]
As it doesn't require auth from the token supplied via the env var

**Reply by mikael-andersson** [reply]
@sinclairpaul Can I make it work by configuring (via UI) the SSH add-on in some way? Would I need to create my own SSH add-on to get it to work? What tweak(s) would I need for a customized SSH add-on to work? Or how can I accomplish this in any other way? If you do not have time to answer, can you please send me a link with relevant information regarding this?

**Reply by mikael-andersson** [reply]
OK, I did not first get what you talked about the token, but I think I understand now.

**Reply by mikael-andersson** [reply]
Can I configure the SSH add-on in order to get hold on the env token? Or do I need to know the token and pass it with my command?

**Reply by sinclairpaul** [reply]
Here you go ```ssh user@myaddress "zsh --login -c 'ha core info'"```

**Reply by mikael-andersson** [reply]
That was easy. :) I tried it with bash, it works! :) Many thanks. Can I donate money to you?

**Reply by sinclairpaul** [reply]
No need, glad you got it sorted:)

**Reply by mikael-andersson** [reply]
Ok, thanks. Can I ask another thing: when I add 'username: testuser' to the configuration, it seems that the user does not get added after restarting the add-on. I do not see anything in the log about it and when I login, the root user is still the valid user. If you do not know anything about it, I will create an issue. I need to sleep now. :) Thanks again.

**Reply by sinclairpaul** [reply]
Should be overwritten, but nope sorry no solid answer.

**Reply by mikael-andersson** [reply]
I will try to look into it more, thanks.
